src/layouts/Fields/styles/FieldRow.js

```diff
@@ -8,33 +8,66 @@ const copyArea = (input, times) =>
-  <div
-    style={{
-      display: 'grid',
-      // each row has 3 sub-rows: label, content, helpText.
-      // there's a named area for each column.
-      gridTemplateAreas: [
-        `"${copyArea('label', columns)}"`,
-        `"${copyArea('content', columns)}"`,
-        `"${copyArea('helpText', columns)}"`,
-      ].join('\n'),
-      // defines spacing between rows and columns
-      gridGap: `${themeGet('spacing.extraSmall')({ theme })} ${themeGet(
-        'spacing.large',
-      )({ theme })}`,
-      // each column has an equal size. To make fields larger than
-      // adjacent fields, use the columnSpan prop on Field to span
-      // multiple columns. This keeps all field sizes directly proportional
-      gridTemplateColumns: `repeat(${columns}, 1fr)`,
-      // label and helpText rows expand or contract as needed.
-      // content row has a minimum size of 53px, and can expand as needed.
-      gridTemplateRows: 'auto minmax(53px, auto) auto',
-    }}
-  >
-    {children}
-  </div>
-);
+const hasLabel = children =>
+  React.Children.toArray(children).some(
+    child =>
+      child.props.Label != null &&
+      child.props.label != null &&
+      child.props.label.length > 0,
+  );
+
+const hasHelpText = children =>
+  React.Children.toArray(children).some(
+    child =>
+      child.props.HelpText != null &&
+      child.props.helpText != null &&
+      child.props.helpText.length > 0,
+  );
+
+/*
+ * We calculate whether we need the label and help text rows based on
+ * whether any children need them. This helps deal with the gridGap,
+ * as it will still be present if a grid area gets a size of zero.
+ */
+const calcGridTemplateAreas = (children, columns) => {
+  const areas = [];
+  if (hasLabel(children)) {
+    areas.push(copyArea('label', columns));
+  }
+  areas.push(copyArea('content', columns));
+  if (hasHelpText(children)) {
+    areas.push(copyArea('helpText', columns));
+  }
+  return areas.map(a => `"${a}"`).join('\n');
+};
+
+const FieldRow = ({ children, columns, theme }) => {
+  return (
+    <div
+      style={{
+        display: 'grid',
+        // each row has 3 sub-rows: label, content, helpText.
+        // there's a named area for each column.
+        gridTemplateAreas: calcGridTemplateAreas(children, columns),
+        // defines spacing between rows and columns
+        gridGap: `${themeGet('spacing.extraSmall')({ theme })} ${themeGet(
+          'spacing.large',
+        )({ theme })}`,
+        // each column has an equal size. To make fields larger than
+        // adjacent fields, use the columnSpan prop on Field to span
+        // multiple columns. This keeps all field sizes directly proportional
+        gridTemplateColumns: `repeat(${columns}, 1fr)`,
+        // label and helpText rows expand or contract as needed.
+        // content row has a minimum size of 53px, and can expand as needed.
+        gridTemplateRows: `${
+          hasLabel(children) ? 'auto' : ''
```

Comment:
This logic starts to get a little confusing, since the logic here must align with the logic for creating template areas. I'd almost rather abandon the use of `grid-gap` altogether in CSS and move to include the spacing within the label/helpText elements themselves to avoid all this conditional processing. What do you think?

Reply:
I don't think that would work too well for two reasons:
- When creating gaps in a series of items, you can use `:last-child` to make sure the final one doesn't add more space, but that isn't going to work in two dimensions very well.
- Instead, we could use gaps between fields but consider the full label + content + help text to be a single field. Part of the benefit we get out of css grid is making sure everything lines up very consistently. We could probably do it by having conditional spacing I guess.

I might be able to clean up the logic further - I can use one function to decide which areas should be present, then divert to different functions to map them to strings.